web/src/lib/components/shared-components/profile-image-cropper.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+  import { AssetResponseDto, api } from '@api';
+  import { createEventDispatcher } from 'svelte';
+  import PhotoViewer from '../asset-viewer/photo-viewer.svelte';
+  import BaseModal from './base-modal.svelte';
+  import Button from '../elements/buttons/button.svelte';
+  const dispatch = createEventDispatcher();
+  export let asset: AssetResponseDto;
+  export let publicSharedKey = '';
+  import domtoimage from 'dom-to-image';
+  import { notificationController, NotificationType } from './notification/notification';
+
+  let profilePicture: HTMLDivElement;
+
+  const handleSetProfilePicture = async () => {
+    const div = profilePicture.childNodes[0] as HTMLDivElement;
+    const blob: Blob = await domtoimage.toBlob(div);
+    const file: File = new File([blob], 'profile-picture.png', { type: 'image/png' });
+    try {
+      await api.userApi.createProfileImage({ file });
+      dispatch('close');
+      notificationController.show({
+        type: NotificationType.Info,
+        message: 'Profile picture set.',
+        timeout: 3000,
+      });
+    } catch (err) {
+      console.error('Error [profile-image-cropper]:', err);
+      notificationController.show({
+        type: NotificationType.Error,
+        message: 'Error setting profile picture.',
+        timeout: 3000,
+      });
+    }
+  };
+</script>
+
+<BaseModal on:close={() => dispatch('close')}>
```

Comment:
Can you use bind:close instead?

Reply:
I just copied from album-selection-model.svelte. How would bind:close be used?

Reply:
Sorry, I was thinking of `on:click` which I believe is shorthand for this. If the implementation just emits the event with the same name again. Does that work here?